internals/enterprise_helpers.py

```diff
@@ -0,0 +1,148 @@
+
+  # We are creating a new feature
+  if existing_feature is None:
+    # All enterprise features need this
+    if new_fields['feature_type'] == FEATURE_TYPE_ENTERPRISE_ID:
+      return not has_valid_milestone_in_new_fields
+    # All breaking changes need this
+    if 'breaking_change' in new_fields and new_fields['breaking_change'] == True:
+      return not has_valid_milestone_in_new_fields
+    return False
+
+  # We are updating a feature that already has a notification milestone
+  if existing_feature.first_enterprise_notification_milestone != None:
+    return False
+  
+  # The enterprise feature we are updating does not have the field
+  if existing_feature.feature_type == FEATURE_TYPE_ENTERPRISE_ID:
+    return not has_valid_milestone_in_new_fields
+  
+  # The breaking change stays a breaking change
+  if existing_feature.breaking_change == True and ('breaking_change' not in new_fields or
+                                                    new_fields['breaking_change'] == True):
+    return not has_valid_milestone_in_new_fields
+  
+  # The normal feature becomes a breaking change
+  if 'breaking_change' in new_fields and new_fields['breaking_change'] == True:
+    return not has_valid_milestone_in_new_fields
+  
+  return False
+
+def can_update_first_notification_milestone(feature: FeatureEntry, new_fields: dict) -> bool:
+  """Returns whether the milestone can be used to update first_enterprise_notification_milestone.
+
+  Args:
+    existing_feature: FeatureEntry feature that needs to be updated.
+    new_fields: dict Fields that will be used to update or create the feature.
+  """
+  if ('first_enterprise_notification_milestone' not in new_fields or
+      not new_fields['first_enterprise_notification_milestone']):
+    return False
+  milestone = new_fields['first_enterprise_notification_milestone']
+
+  # We cannot set a milestone that has already been released
+  milestone_details = channels_api.construct_specified_milestones_details(milestone, milestone)
+  if (milestone not in milestone_details or
+      _str_to_datetime(milestone_details[milestone]['stable_date']) <= datetime.now()):
+    return False
+  
+  # We cannot update the milestone of a feature that has already been announced
+  if feature.first_enterprise_notification_milestone != None:
+    milestone = feature.first_enterprise_notification_milestone
+    previous_milestone_details = channels_api.construct_specified_milestones_details(milestone,
+                                                                                     milestone)
+    if (milestone in previous_milestone_details and
+        _str_to_datetime(previous_milestone_details[milestone]['stable_date']) <= datetime.now()):
+      return False
+
+  if feature.feature_type == FEATURE_TYPE_ENTERPRISE_ID:
+    return True
+  
+  # The breaking change stays a breaking change
+  if feature.breaking_change == True and ('breaking_change' not in new_fields 
```

Comment:
This whole condition could be simpler as:
`new_fields.get('breaking_change', feature.breaking_change)`
And, with that, I don't think that you need the next if-statement.